cli/src/commands/git/clone.rs

```diff
@@ -259,7 +266,10 @@ fn fetch_new_remote(
             &[StringPattern::everything()],
             cb,
             depth,
-            FetchTagsOverride::UseRemoteConfiguration,
+            match fetch_tags {
+                FetchTagsMode::All | FetchTagsMode::Included => FetchTagsOverride::ForceAllTags,
+                FetchTagsMode::None => FetchTagsOverride::UseRemoteConfiguration,
+            },
```

Comment:
I think `fetch_tags` should be respected if explicitly specified as `--fetch-tags=included`. We'll need to change the arg type to `Option<FetchTagsMode>`.

Reply:
I like this idea! I reworked this so `--fetch-tags` is now optional when cloning: if specified, we'll both configure the remote with the appropriate setting, and also apply the same behavior for the initial fetch. If _not_ specified, we'll still configure the remote with the default behavior of only fetching included tags, but the initial fetch _will_ fetch all tags.

This also led to me dropping the commits which introduced `RemoteFetchTagsMode`/`CloneFetchTagsMode` since now the documentation between the options can be shared.